portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -175,4 +185,18 @@ private GoogleIdToken validate(String idTokenString) throws Exception {
     }
     return idToken;
   }
+
+  private float sentimentAnalysis(String message) throws IOException {
+    Document doc = Document
+      .newBuilder()
+      .setContent(message)
+      .setType(Document.Type.PLAIN_TEXT)
+      .build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    AnalyzeSentimentResponse response = languageService.analyzeSentiment(doc);
+    Sentiment sentiment = response.getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
```

Comment:
[LanguageServiceClient](https://googleapis.dev/java/google-cloud-language/latest/com/google/cloud/language/v1/LanguageServiceClient.html) implements `AutoCloseable`, so you should use [try-with-resource](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) here.